packages/http-client-csharp/generator/Microsoft.Generator.CSharp/test/Providers/ParameterProviderTests.cs

```diff
@@ -30,6 +38,39 @@ public void Equals(ParameterProvider p1, ParameterProvider? p2, bool areEqual)
             Assert.AreEqual(areEqual, result);
         }
 
+        [TestCaseSource(nameof(DefaultValueTestCases))]
+        public void DefaultValue(InputParameter inputParameter, bool isCollectionType)
+        {
+            var provider = new ParameterProvider(inputParameter);
+            var parsedValue = inputParameter.DefaultValue?.Value;
+
+            if (parsedValue?.GetType() == typeof(int))
```

Comment:
given the big if/else here lets just make a few test cases instead its much easier to follow